### PR TITLE
add soql function magnitude

### PR DESCRIPTION
**Pull request by sokrahta**
Pair CR: @chitang

### CODE REVIEW

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala [hunk 1]
```diff
@@ -79,6 +79,8 @@ object SoQLFunctions {
   val UnaryPlus = Function("unary +", SpecialFunctions.Operator("+"), Map("a"->NumLike), Seq(VariableType("a")), Seq.empty, VariableType("a"))
   val UnaryMinus = Function("unary -", SpecialFunctions.Operator("-"), Map("a"->NumLike), Seq(VariableType("a")), Seq.empty, VariableType("a"))
 
+  val Magnitude = Function("magnitude", FunctionName("magnitude"), Map("a"->NumLike, "b"->NumLike), Seq(VariableType("a"), VariableType("b")), Seq.empty, VariableType("b"))
```
Comment (znep) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala]: Should the result of this function be a SoQLNumber regardless of input type?

Comment (chitang) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala]: I think it should follow the first argument in the soql function - which is "a".  We were confused between which one is the log based.

Comment (sokrahta) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala]: The result is always a positive integer. Similarly the base argument is always a positive integer.

Comment (chitang) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala]: Yes, the result is always positive integer.  But there is value in following the datatype of the primary argument.  Postgres round function does that:

Function: round(dp or numeric)
Return Type: (same as input)
Description: round to nearest integer

Comment (sokrahta) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala]: that's fair, thanks chi

**Pull request merged**